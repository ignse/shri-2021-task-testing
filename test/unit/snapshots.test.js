/**
 * @jest-environment jsdom
 */

import React from 'react';
import renderer from 'react-test-renderer';
import {ProductDetails} from '../../src/client/components/ProductDetails';
import {CartApi, ExampleApi} from '../../src/client/api';
import {initStore} from '../../src/client/store';
import {Provider} from 'react-redux';
import {ProductItem} from '../../src/client/components/ProductItem';
import { BrowserRouter } from 'react-router-dom';
import {CartBadge} from '../../src/client/components/CartBadge';
import {Form} from '../../src/client/components/Form';
import {Contacts} from '../../src/client/pages/Contacts';
import {Delivery} from '../../src/client/pages/Delivery';
import {Home} from '../../src/client/pages/Home';
import {Catalog} from '../../src/client/pages/Catalog';
import {Route} from 'react-router';
import {Cart} from '../../src/client/pages/Cart';

const api = new ExampleApi('');
const cart = new CartApi();
const store = initStore(api, cart);

it('renders correctly ProductDetails', () => {
	const product = {
		id: 2,
		name: 'bbb',
		price: 200,
		description: '',
		material: '',
		color: ''
	};

	const tree = renderer
		.create( <Provider store={store}><ProductDetails product={product} /></Provider>)
		.toJSON();
	expect(tree).toMatchSnapshot();
});

it('renders correctly ProductItem', () => {
	const product = {
		id: 2,
		name: 'bbb',
		price: 200,
		description: '',
		material: '',
		color: ''
	};

	const tree = renderer
		.create(  <BrowserRouter basename={'/'}><Provider store={store}><ProductItem product={product} /></Provider></BrowserRouter>)
		.toJSON();
	expect(tree).toMatchSnapshot();
});

it('renders correctly Form', () => {
	const tree = renderer
		.create(  <Form onSubmit={() => {}} />)
		.toJSON();
	expect(tree).toMatchSnapshot();
});

it('renders correctly Contacts', () => {
	const tree = renderer
		.create( <Contacts />)
		.toJSON();
	expect(tree).toMatchSnapshot();
});

it('renders correctly Delivery', () => {
	const tree = renderer
		.create( <Delivery />)
		.toJSON();
	expect(tree).toMatchSnapshot();
});

it('renders correctly Home', () => {
	const tree = renderer
		.create( <Home />)
		.toJSON();
	expect(tree).toMatchSnapshot();
});

it('renders correctly Catalog', () => {
	const products = [
		{
		id: 111,
		name: 'aaaa',
		description: 'aaaa desc',
		price: 555,
		color: 'black',
		material: 'metal',
		},
		{
			id: 222,
			name: 'bbb',
			description: 'bbb desc',
			price: 1555,
			color: 'white',
			material: 'plastic',
		}
	];

	store.dispatch({type: 'PRODUCTS_LOADED', products: products});

	const tree = renderer
		.create(<BrowserRouter path="/catalog" exact><Provider store={store}><Catalog /></Provider></BrowserRouter>)
		.toJSON();
	expect(tree).toMatchSnapshot();
});