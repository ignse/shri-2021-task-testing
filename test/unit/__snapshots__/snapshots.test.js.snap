// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly Cart 1`] = `
<div
  className="Cart"
>
  <div
    className="row mb-4"
  >
    <div
      className="col"
    >
      <h1>
        Shopping cart
      </h1>
      <table
        className="Cart-Table table"
      >
        <thead>
          <tr>
            <th
              scope="col"
            >
              #
            </th>
            <th
              scope="col"
            >
              Product
            </th>
            <th
              scope="col"
            >
              Price
            </th>
            <th
              scope="col"
            >
              Count
            </th>
            <th
              scope="col"
            >
              Total
            </th>
          </tr>
        </thead>
        <tbody>
          <tr
            data-testid="5"
          >
            <th
              className="Cart-Index"
              scope="row"
            >
              1
            </th>
            <td
              className="Cart-Name"
            >
              Test product
            </td>
            <td
              className="Cart-Price"
            >
              $
              500
            </td>
            <td
              className="Cart-Count"
            >
              2
            </td>
            <td
              className="Cart-Total"
            >
              $
              1000
            </td>
          </tr>
        </tbody>
        <tfoot>
          <tr>
            <td
              colSpan={4}
            >
              Order price:
            </td>
            <td
              className="Cart-OrderPrice"
            >
              $
              1000
            </td>
          </tr>
        </tfoot>
      </table>
    </div>
  </div>
  <div
    className="row mb-4"
  >
    <div
      className="col-6"
    >
      <button
        className="Cart-Clear btn btn-outline-secondary"
        onClick={[Function]}
      >
        Clear shopping cart
      </button>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col-12 col-sm-8 col-md-6 col-lg-4"
    >
      <h2>
        Ð¡heckout
      </h2>
      <div
        className="Form"
      >
        <div
          className="mb-3"
        >
          <label
            className="form-label"
            htmlFor="f-name"
          >
            Name
          </label>
          <input
            autoComplete="off"
            className="Form-Field Form-Field_type_name form-control"
            disabled={false}
            id="f-name"
            onChange={[Function]}
            type="text"
          />
          <div
            className="invalid-feedback"
          >
            Please provide your name
          </div>
        </div>
        <div
          className="mb-3"
        >
          <label
            className="form-label"
            htmlFor="f-phone"
          >
            Phone
          </label>
          <input
            className="Form-Field Form-Field_type_phone form-control"
            disabled={false}
            id="f-phone"
            onChange={[Function]}
            type="text"
          />
          <div
            className="invalid-feedback"
          >
            Please provide a valid phone
          </div>
        </div>
        <div
          className="mb-3"
        >
          <label
            className="form-label"
            htmlFor="f-address"
          >
            Address
          </label>
          <textarea
            className="Form-Field Form-Field_type_address form-control"
            disabled={false}
            id="f-address"
            onChange={[Function]}
            rows={3}
          />
          <div
            className="invalid-feedback"
          >
            Please provide a valid address
          </div>
        </div>
        <button
          className="Form-Submit btn btn-primary"
          disabled={false}
          onClick={[Function]}
        >
          Checkout
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`renders correctly Catalog 1`] = `
<div
  className="Catalog"
>
  <div
    className="row"
  >
    <div
      className="col"
    >
      <h1>
        Catalog
      </h1>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col-12 col-sm-6 col-md-4 col-lg-3"
      data-testid={111}
    >
      <div
        className="ProductItem card w-100 mb-4"
        data-testid={111}
      >
        <img
          className="Image card-img-top"
          src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAQAAAC1HAwCAAAAC0lEQVR42mNkMAYAADkANVKH3ScAAAAASUVORK5CYII="
        />
        <div
          className="card-body"
        >
          <h5
            className="ProductItem-Name card-title"
          >
            aaaa
          </h5>
          <p
            className="ProductItem-Price card-text"
          >
            $
            555
          </p>
          <a
            className="ProductItem-DetailsLink card-link"
            href="/catalog/111"
            onClick={[Function]}
          >
            Details
          </a>
        </div>
      </div>
    </div>
    <div
      className="col-12 col-sm-6 col-md-4 col-lg-3"
      data-testid={222}
    >
      <div
        className="ProductItem card w-100 mb-4"
        data-testid={222}
      >
        <img
          className="Image card-img-top"
          src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAQAAAC1HAwCAAAAC0lEQVR42mNkMAYAADkANVKH3ScAAAAASUVORK5CYII="
        />
        <div
          className="card-body"
        >
          <h5
            className="ProductItem-Name card-title"
          >
            bbb
          </h5>
          <p
            className="ProductItem-Price card-text"
          >
            $
            1555
          </p>
          <a
            className="ProductItem-DetailsLink card-link"
            href="/catalog/222"
            onClick={[Function]}
          >
            Details
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`renders correctly Contacts 1`] = `
<div
  className="Contacts"
>
  <div
    className="row"
  >
    <div
      className="col"
    >
      <h1>
        Contacts
      </h1>
      <p>
        Ut non consequatur aperiam ex dolores. Voluptatum harum consequatur est totam. Aut voluptatum aliquid aut optio et ea. Quaerat et eligendi minus quasi. Culpa voluptatem voluptatem dolores molestiae aut quos iure. Repellat aperiam ut aliquam iure. Veritatis magnam quisquam et dolorum recusandae aut.
      </p>
      <p>
        Molestias inventore illum architecto placeat molestias ipsam facilis ab quo. Rem dolore cum qui est reprehenderit assumenda voluptatem nisi ipsa. Unde libero quidem. Excepturi maiores vel quia. Neque facilis nobis minus veniam id. Eum cum eveniet accusantium molestias voluptas aut totam laborum aut. Ea molestiae ullam et. Quis ea ipsa culpa eligendi ab sit ea error suscipit. Quia ea ut minus distinctio quam eveniet nihil. Aut voluptate numquam ipsa dolorem et quas nemo.
      </p>
    </div>
  </div>
</div>
`;

exports[`renders correctly Delivery 1`] = `
<div
  className="Delivery"
>
  <div
    className="row"
  >
    <div
      className="col"
    >
      <h1>
        Delivery
      </h1>
      <p>
        Deserunt occaecati tempora. Qui occaecati est aliquam. Enim qui nulla ipsam. Incidunt impedit enim consequuntur amet at consequuntur vero. Dolor et ad facere asperiores iste est praesentium quaerat iure. Quibusdam mollitia autem quos voluptas quia est doloremque corporis et. Sed fuga quasi esse perspiciatis fugit maxime. Qui quidem amet.
      </p>
      <img
        className="Image w-25 mb-4"
        src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAQAAAC1HAwCAAAAC0lEQVR42mNkMAYAADkANVKH3ScAAAAASUVORK5CYII="
      />
      <p>
        Dolores magnam consequatur iste aliquam qui sint non ab. Culpa saepe omnis. Recusandae vel aperiam voluptates harum. Perspiciatis qui molestiae qui tempora quisquam. Mollitia voluptatum minus laboriosam. Dolor maiores possimus repudiandae praesentium hic eos. Veritatis et repellat.
      </p>
      <p>
        Pariatur nisi nobis hic ut facilis sunt rerum id error. Soluta nihil quisquam quia rerum illo. Ipsam et suscipit est iure incidunt quasi et eum. Culpa libero dignissimos recusandae. In magni sapiente non voluptas molestias. Deserunt quos quo placeat sunt. Ea necessitatibus dolores eaque ex aperiam sunt eius. Saepe aperiam aut. Quaerat natus consequatur aut est id saepe et aut facilis.
      </p>
    </div>
  </div>
</div>
`;

exports[`renders correctly Form 1`] = `
<div
  className="Form"
>
  <div
    className="mb-3"
  >
    <label
      className="form-label"
      htmlFor="f-name"
    >
      Name
    </label>
    <input
      autoComplete="off"
      className="Form-Field Form-Field_type_name form-control"
      disabled={false}
      id="f-name"
      onChange={[Function]}
      type="text"
    />
    <div
      className="invalid-feedback"
    >
      Please provide your name
    </div>
  </div>
  <div
    className="mb-3"
  >
    <label
      className="form-label"
      htmlFor="f-phone"
    >
      Phone
    </label>
    <input
      className="Form-Field Form-Field_type_phone form-control"
      disabled={false}
      id="f-phone"
      onChange={[Function]}
      type="text"
    />
    <div
      className="invalid-feedback"
    >
      Please provide a valid phone
    </div>
  </div>
  <div
    className="mb-3"
  >
    <label
      className="form-label"
      htmlFor="f-address"
    >
      Address
    </label>
    <textarea
      className="Form-Field Form-Field_type_address form-control"
      disabled={false}
      id="f-address"
      onChange={[Function]}
      rows={3}
    />
    <div
      className="invalid-feedback"
    >
      Please provide a valid address
    </div>
  </div>
  <button
    className="Form-Submit btn btn-primary"
    disabled={false}
    onClick={[Function]}
  >
    Checkout
  </button>
</div>
`;

exports[`renders correctly Home 1`] = `
<div
  className="Home"
>
  <div
    className="row"
  >
    <div
      className="col bg-secondary text-white py-4 bg-opacity-75"
    >
      <p
        className="display-3"
      >
        Welcome to Example store!
      </p>
      <p
        className="lead"
      >
        Culpa perspiciatis corporis facilis fugit similique
      </p>
      <p
        className="lead"
      >
        Cum aut ut eveniet rem cupiditate natus veritatis quia
      </p>
    </div>
  </div>
  <div
    className="row mb-4"
  >
    <div
      className="col-12 col-md-4 bg-light py-3"
    >
      <h1>
        Quickly
      </h1>
      <p
        className="lead"
      >
        Odio aut assumenda ipsam amet reprehenderit. Perspiciatis qui molestiae qui tempora quisquam
      </p>
    </div>
    <div
      className="col-12 col-md-4 bg-light py-3"
    >
      <h1>
        Qualitatively
      </h1>
      <p
        className="lead"
      >
        Ut nisi distinctio est non voluptatem. Odio aut assumenda ipsam amet reprehenderit
      </p>
    </div>
    <div
      className="col-12 col-md-4 bg-light py-3"
    >
      <h1>
        Inexpensive
      </h1>
      <p
        className="lead"
      >
        Perspiciatis qui molestiae qui tempora quisquam. Ut nisi distinctio est non voluptatem
      </p>
    </div>
  </div>
  <div
    className="row mb-4"
  >
    <div
      className="col-12py-3"
    >
      <p>
        Sed voluptatum quis voluptates laudantium incidunt laudantium. Illo non quos eos vel ipsa. Explicabo itaque est optio neque rerum provident enim qui sed. Corrupti commodi voluptatem vero soluta hic.
      </p>
      <p>
        Modi corporis consectetur aliquid sit cum tenetur enim. Sed voluptatum quis voluptates laudantium incidunt laudantium. Illo non quos eos vel ipsa. Explicabo itaque est optio neque rerum provident enim qui sed. Corrupti commodi voluptatem vero soluta hic.
      </p>
    </div>
  </div>
</div>
`;

exports[`renders correctly ProductDetails 1`] = `
<div
  className="ProductDetails row"
>
  <div
    className="col-12 col-sm-5 col-lg-4"
  >
    <img
      className="Image"
      src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAQAAAC1HAwCAAAAC0lEQVR42mNkMAYAADkANVKH3ScAAAAASUVORK5CYII="
    />
  </div>
  <div
    className="col-12 col-sm-7 col-lg-6"
  >
    <h1
      className="ProductDetails-Name"
    >
      bbb
    </h1>
    <p
      className="ProductDetails-Description"
    >
      
    </p>
    <p
      className="ProductDetails-Price fs-3"
    >
      $
      200
    </p>
    <p>
      <button
        className="ProductDetails-AddToCart btn btn-primary btn-lg"
        onClick={[Function]}
      >
        Add to Cart
      </button>
    </p>
    <dl>
      <dt>
        Color
      </dt>
      <dd
        className="ProductDetails-Color text-capitalize"
      >
        
      </dd>
      <dt>
        Material
      </dt>
      <dd
        className="ProductDetails-Material text-capitalize"
      >
        
      </dd>
    </dl>
  </div>
</div>
`;

exports[`renders correctly ProductItem 1`] = `
<div
  className="ProductItem card w-100 mb-4"
  data-testid={2}
>
  <img
    className="Image card-img-top"
    src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAQAAAC1HAwCAAAAC0lEQVR42mNkMAYAADkANVKH3ScAAAAASUVORK5CYII="
  />
  <div
    className="card-body"
  >
    <h5
      className="ProductItem-Name card-title"
    >
      bbb
    </h5>
    <p
      className="ProductItem-Price card-text"
    >
      $
      200
    </p>
    <a
      className="ProductItem-DetailsLink card-link"
      href="/catalog/2"
      onClick={[Function]}
    >
      Details
    </a>
  </div>
</div>
`;
